modules/sdk-coin-algo/src/algo.ts

```diff
@@ -188,14 +187,24 @@ export class Algo extends BaseCoin {
     };
   }
 
+  /** inheritdoc */
+  generateRootKeyPair(seed?: Buffer): KeyPair {
+    const keyPair = seed ? new AlgoLib.KeyPair({ seed }) : new AlgoLib.KeyPair();
+    const keys = keyPair.getKeys();
+    if (!keys.prv) {
+      throw new Error('Missing prv in key generation.');
+    }
+    return { prv: keys.prv + keys.pub, pub: keys.pub };
```

Comment:
Does the prv need the pub to be appended? 🤔

Reply:
Not clear on how/why the return values are different for `generateKeyPair` and `generateRootKeyPair` even though the return type is the same